File: twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if(len(nums) < 2): 
            return nums
        dict = {nums[0] : 0}
        for i in range(1,len(nums)):
            checkNum = target - nums[i]
            if(checkNum in dict.keys()):
                return [i, dict[checkNum]]
            else:
                dict[nums[i]] = i
        

File: easy/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // key = number needed to form sum, value = index of first number in sum
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(nums[i])) {
                return new int[] {map.get(nums[i]), i};
            }
            map.put(target - nums[i], i);
        }
        return null;
    }
}